torch/lib/c10d/CMakeLists.txt

```diff
@@ -1,8 +1,34 @@
 cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake ${CMAKE_MODULE_PATH})
 
+# Relies on CMAKE_INSTALL_PREFIX to be set to ../tmp_install.
+# It then finds $PREFIX/share/cmake/ATen/ATenConfig.cmake,
+# which defines ATEN_INCLUDE_DIR and ATEN_LIBRARIES.
+find_package(ATen REQUIRED)
+if(NOT ATen_FOUND)
+  message(FATAL_ERROR "ATen not found")
+endif()
+
+find_package(Gloo REQUIRED)
+if(NOT Gloo_FOUND)
+  message(FATAL_ERROR "Gloo not found")
+endif()
+
 add_library(store Utils.cpp Store.cpp FileStore.cpp TCPStore.cpp)
 target_compile_options(store PUBLIC "-std=c++11")
+target_include_directories(store PUBLIC ${ATEN_INCLUDE_DIR})
+
+add_library(process_group ProcessGroup.cpp)
+target_compile_options(process_group PUBLIC "-std=c++11")
+target_include_directories(process_group PUBLIC ${ATEN_INCLUDE_DIR})
+target_link_libraries(process_group PUBLIC ${ATEN_LIBRARIES})
+
+add_library(process_group_gloo ProcessGroupGloo.cpp)
```

Comment:
Huh this is quite surprising. Do we really want to scatted c10d into so many distinct libraries? I think it's ok to have a separate one for each backend, but why is store and process_group separate?

Reply:
Agree, can be cleaned up.

Reply:
Fixed. Now we have c10d and c10d_gloo targets.